spec/lib/parse_spec.rb

```diff
@@ -1534,16 +1534,52 @@
 
     it 'finds tables referenced in the FROM clause' do
       query = described_class.parse(<<-SQL)
-        UPDATE users SET name = users_new.name FROM users_new WHERE users.id = users_new.id
+        UPDATE users SET name = users_new.name
+        FROM users_new
+        INNER JOIN join_table ON join_table.user_id = new_users.id
+        WHERE users.id = users_new.id
       SQL
       expect(query.warnings).to be_empty
-      expect(query.tables).to eq(['users', 'users_new'])
-      expect(query.select_tables).to eq(['users_new'])
+      expect(query.tables).to eq(['users', 'users_new', 'join_table'])
+      expect(query.select_tables).to eq(['users_new', 'join_table'])
```

Comment:
This spec change is unrelated to the main change of the PR. I think the spec as written in this branch would have failed in a slightly older version of the code but now it is passing.